In [1]:
import pandas as pd
import numpy as np

import matplotlib as mpl
import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings('ignore')

pd.set_option('display.max_columns', None)

mpl.rcParams['axes.unicode_minus'] = False
plt.rcParams['font.family'] = 'Malgun Gothic'


pd.set_option('display.max_rows', 500)
pd.set_option('display.max_columns', 500)

In [2]:
df = pd.read_csv('./1년 1시간별 Ls, Ws.csv', encoding = 'cp949')
df

자치구           대여소명  대여소번호     년  월  시간    lambda        mu    c  \
0       중랑구  (구)신한은행 중랑교지점   1442  2020  6   0  3.800000  2.700000  6.0   
1       중랑구  (구)신한은행 중랑교지점   1442  2020  6   1  3.700000  2.133333  6.0   
2       중랑구  (구)신한은행 중랑교지점   1442  2020  6   2  2.133333  1.266667  6.0   
3       중랑구  (구)신한은행 중랑교지점   1442  2020  6   3  1.566667  0.833333  6.0   
4       중랑구  (구)신한은행 중랑교지점   1442  2020  6   4  1.333333  0.533333  6.0   
...     ...            ...    ...   ... ..  ..       ...       ...  ...   
694939  강서구        힐스테이트에코   2720  2021  5  19  7.933333  8.066667  9.0   
694940  강서구        힐스테이트에코   2720  2021  5  20  5.166667  7.800000  9.0   
694941  강서구        힐스테이트에코   2720  2021  5  21  3.400000  6.933333  9.0   
694942  강서구        힐스테이트에코   2720  2021  5  22  1.800000  3.000000  9.0   
694943  강서구        힐스테이트에코   2720  2021  5  23  1.233333  1.166667  9.0   

             rho       p_0       p_1      p_1+       p_c       L_s        W_s  
0       1.407407  0.040995  0.057696  0.959005  0.318602  4.249820   1.641292  
1       1.734375  0.015893  0.027565  0.984107  0.432587  4.789792   2.281475  
2       1.684211  0.018275  0.030779  0.981725  0.417101  4.725431   3.800050  
3       1.880000  0.010731  0.020174  0.989269  0.473793  4.948997   6.003217  
4       2.500000  0.002462  0.006154  0.997538  0.600985  5.344821  10.046270  
...          ...       ...       ...       ...       ...       ...        ...  
694939  0.983471  0.107665  0.105886  0.892335  0.092668  4.362561   0.606066  
694940  0.662393  0.343188  0.227325  0.656812  0.008425  1.796723   0.350708  
694941  0.490385  0.510026  0.250109  0.489974  0.000836  0.954216   0.280887  
694942  0.600000  0.402433  0.241460  0.597567  0.004056  1.439166   0.802792  
694943  1.057143  0.076892  0.081286  0.923108  0.126790  4.956086   4.601924  

[694944 rows x 16 columns]

In [3]:
df = df[df['대여소명'] == '올림픽공원역 3번출구']
df

자치구         대여소명  대여소번호     년   월  시간     lambda         mu     c  \
506592  송파구  올림픽공원역 3번출구   2622  2020   6   0   2.500000   2.000000  18.0   
506593  송파구  올림픽공원역 3번출구   2622  2020   6   1   1.033333   1.166667  18.0   
506594  송파구  올림픽공원역 3번출구   2622  2020   6   2   0.500000   0.533333  18.0   
506595  송파구  올림픽공원역 3번출구   2622  2020   6   3   0.433333   0.333333  18.0   
506596  송파구  올림픽공원역 3번출구   2622  2020   6   4   0.133333   0.100000  18.0   
506597  송파구  올림픽공원역 3번출구   2622  2020   6   5   0.733333   0.333333  18.0   
506598  송파구  올림픽공원역 3번출구   2622  2020   6   6   3.866667   1.900000  18.0   
506599  송파구  올림픽공원역 3번출구   2622  2020   6   7   4.366667   3.733333  18.0   
506600  송파구  올림픽공원역 3번출구   2622  2020   6   8   7.000000   6.900000  18.0   
506601  송파구  올림픽공원역 3번출구   2622  2020   6   9   6.066667   5.866667  18.0   
506602  송파구  올림픽공원역 3번출구   2622  2020   6  10   5.033333   5.900000  18.0   
506603  송파구  올림픽공원역 3번출구   2622  2020   6  11   5.933333   6.266667  18.0   
506604  송파구  올림픽공원역 3번출구   2622  2020   6  12   7.400000   6.466667  18.0   
506605  송파구  올림픽공원역 3번출구   2622  2020   6  13   6.766667   8.033333  18.0   
506606  송파구  올림픽공원역 3번출구   2622  2020   6  14   6.366667   6.600000  18.0   
506607  송파구  올림픽공원역 3번출구   2622  2020   6  15   7.633333   8.733333  18.0   
506608  송파구  올림픽공원역 3번출구   2622  2020   6  16  11.266667  11.500000  18.0   
506609  송파구  올림픽공원역 3번출구   2622  2020   6  17  14.600000  16.200000  18.0   
506610  송파구  올림픽공원역 3번출구   2622  2020   6  18  16.366667  17.366667  18.0   
506611  송파구  올림픽공원역 3번출구   2622  2020   6  19  17.600000  18.533333  18.0   
506612  송파구  올림픽공원역 3번출구   2622  2020   6  20  21.900000  21.333333  18.0   
506613  송파구  올림픽공원역 3번출구   2622  2020   6  21  17.233333  18.333333  18.0   
506614  송파구  올림픽공원역 3번출구   2622  2020   6  22  12.533333  12.966667  18.0   
506615  송파구  올림픽공원역 3번출구   2622  2020   6  23   4.633333   5.000000  18.0   
506616  송파구  올림픽공원역 3번출구   2622  2020   7   0   1.400000   1.300000  18.0   
506617  송파구  올림픽공원역 3번출구   2622  2020   7   1   0.700000   0.700000  18.0   
506618  송파구  올림픽공원역 3번출구   2622  2020   7   2   0.433333   0.333333  18.0   
506619  송파구  올림픽공원역 3번출구   2622  2020   7   3   0.266667   0.466667  18.0   
506620  송파구  올림픽공원역 3번출구   2622  2020   7   4   0.166667   0.166667  18.0   
506621  송파구  올림픽공원역 3번출구   2622  2020   7   5   0.466667   0.266667  18.0   
506622  송파구  올림픽공원역 3번출구   2622  2020   7   6   3.333333   1.200000  18.0   
506623  송파구  올림픽공원역 3번출구   2622  2020   7   7   4.166667   3.533333  18.0   
506624  송파구  올림픽공원역 3번출구   2622  2020   7   8   4.933333   4.933333  18.0   
506625  송파구  올림픽공원역 3번출구   2622  2020   7   9   4.600000   5.433333  18.0   
506626  송파구  올림픽공원역 3번출구   2622  2020   7  10   4.100000   4.300000  18.0   
506627  송파구  올림픽공원역 3번출구   2622  2020   7  11   4.400000   4.500000  18.0   
506628  송파구  올림픽공원역 3번출구   2622  2020   7  12   5.966667   6.033333  18.0   
506629  송파구  올림픽공원역 3번출구   2622  2020   7  13   6.666667   6.666667  18.0   
506630  송파구  올림픽공원역 3번출구   2622  2020   7  14   7.300000   8.100000  18.0   
506631  송파구  올림픽공원역 3번출구   2622  2020   7  15   8.300000   9.300000  18.0   
506632  송파구  올림픽공원역 3번출구   2622  2020   7  16   9.566667   8.666667  18.0   
506633  송파구  올림픽공원역 3번출구   2622  2020   7  17  10.233333  11.766667  18.0   
506634  송파구  올림픽공원역 3번출구   2622  2020   7  18  12.766667  13.100000  18.0   
506635  송파구  올림픽공원역 3번출구   2622  2020   7  19  13.266667  13.833333  18.0   
506636  송파구  올림픽공원역 3번출구   2622  2020   7  20  15.933333  15.700000  18.0   
506637  송파구  올림픽공원역 3번출구   2622  2020   7  21  14.200000  14.800000  18.0   
506638  송파구  올림픽공원역 3번출구   2622  2020   7  22   8.733333   9.800000  18.0   
506639  송파구  올림픽공원역 3번출구   2622  2020   7  23   5.133333   5.100000  18.0   
506640  송파구  올림픽공원역 3번출구   2622  2020   8   0   0.966667   0.900000  18.0   
506641  송파구  올림픽공원역 3번출구   2622  2020   8   1   0.833333   0.700000  18.0   
506642  송파구  올림픽공원역 3번출구   2622  2020   8   2   0.200000   0.200000  18.0   
5066

In [4]:
df = df.drop(['대여소번호', 'c', 'rho', 'p_c', 'p_0', 'p_1'], axis = 1)
df

자치구         대여소명     년   월  시간     lambda         mu      p_1+  \
506592  송파구  올림픽공원역 3번출구  2020   6   0   2.500000   2.000000  0.996344   
506593  송파구  올림픽공원역 3번출구  2020   6   1   1.033333   1.166667  0.873062   
506594  송파구  올림픽공원역 3번출구  2020   6   2   0.500000   0.533333  0.911549   
506595  송파구  올림픽공원역 3번출구  2020   6   3   0.433333   0.333333  0.997934   
506596  송파구  올림픽공원역 3번출구  2020   6   4   0.133333   0.100000  0.998585   
506597  송파구  올림픽공원역 3번출구  2020   6   5   0.733333   0.333333  1.000000   
506598  송파구  올림픽공원역 3번출구  2020   6   6   3.866667   1.900000  0.999999   
506599  송파구  올림픽공원역 3번출구  2020   6   7   4.366667   3.733333  0.990896   
506600  송파구  올림픽공원역 3번출구  2020   6   8   7.000000   6.900000  0.953905   
506601  송파구  올림픽공원역 3번출구  2020   6   9   6.066667   5.866667  0.961724   
506602  송파구  올림픽공원역 3번출구  2020   6  10   5.033333   5.900000  0.845559   
506603  송파구  올림픽공원역 3번출구  2020   6  11   5.933333   6.266667  0.917663   
506604  송파구  올림픽공원역 3번출구  2020   6  12   7.400000   6.466667  0.987928   
506605  송파구  올림픽공원역 3번출구  2020   6  13   6.766667   8.033333  0.836031   
506606  송파구  올림픽공원역 3번출구  2020   6  14   6.366667   6.600000  0.928628   
506607  송파구  올림픽공원역 3번출구  2020   6  15   7.633333   8.733333  0.863468   
506608  송파구  올림픽공원역 3번출구  2020   6  16  11.266667  11.500000  0.937102   
506609  송파구  올림픽공원역 3번출구  2020   6  17  14.600000  16.200000  0.885337   
506610  송파구  올림픽공원역 3번출구  2020   6  18  16.366667  17.366667  0.914812   
506611  송파구  올림픽공원역 3번출구  2020   6  19  17.600000  18.533333  0.919470   
506612  송파구  올림픽공원역 3번출구  2020   6  20  21.900000  21.333333  0.958856   
506613  송파구  올림픽공원역 3번출구  2020   6  21  17.233333  18.333333  0.913217   
506614  송파구  올림픽공원역 3번출구  2020   6  22  12.533333  12.966667  0.929757   
506615  송파구  올림픽공원역 3번출구  2020   6  23   4.633333   5.000000  0.904107   
506616  송파구  올림픽공원역 3번출구  2020   7   0   1.400000   1.300000  0.975090   
506617  송파구  올림픽공원역 3번출구  2020   7   1   0.700000   0.700000  0.947368   
506618  송파구  올림픽공원역 3번출구  2020   7   2   0.433333   0.333333  0.997934   
506619  송파구  올림픽공원역 3번출구  2020   7   3   0.266667   0.466667  0.571418   
506620  송파구  올림픽공원역 3번출구  2020   7   4   0.166667   0.166667  0.947368   
506621  송파구  올림픽공원역 3번출구  2020   7   5   0.466667   0.266667  0.999982   
506622  송파구  올림픽공원역 3번출구  2020   7   6   3.333333   1.200000  1.000000   
506623  송파구  올림픽공원역 3번출구  2020   7   7   4.166667   3.533333  0.991828   
506624  송파구  올림픽공원역 3번출구  2020   7   8   4.933333   4.933333  0.947368   
506625  송파구  올림픽공원역 3번출구  2020   7   9   4.600000   5.433333  0.839855   
506626  송파구  올림픽공원역 3번출구  2020   7  10   4.100000   4.300000  0.921886   
506627  송파구  올림픽공원역 3번출구  2020   7  11   4.400000   4.500000  0.936056   
506628  송파구  올림픽공원역 3번출구  2020   7  12   5.966667   6.033333  0.941941   
506629  송파구  올림픽공원역 3번출구  2020   7  13   6.666667   6.666667  0.947368   
506630  송파구  올림픽공원역 3번출구  2020   7  14   7.300000   8.100000  0.885337   
506631  송파구  올림픽공원역 3번출구  2020   7  15   8.300000   9.300000  0.878479   
506632  송파구  올림픽공원역 3번출구  2020   7  16   9.566667   8.666667  0.981239   
506633  송파구  올림픽공원역 3번출구  2020   7  17  10.233333  11.766667  0.859811   
506634  송파구  올림픽공원역 3번출구  2020   7  18  12.766667  13.100000  0.934284   
506635  송파구  올림픽공원역 3번출구  2020   7  19  13.266667  13.833333  0.925287   
506636  송파구  올림픽공원역 3번출구  2020   7  20  15.933333  15.700000  0.954063   
506637  송파구  올림픽공원역 3번출구  2020   7  21  14.200000  14.800000  0.925543   
506638  송파구  올림픽공원역 3번출구  2020   7  22   8.733333   9.800000  0.877432   
506639  송파구  올림픽공원역 3번출구  2020   7  23   5.133333   5.100000  0.950397   
506640  송파구  올림픽공원역 3번출구  2020   8   0   0.966667   0.900000  0.974345   
506641  송파구  올림픽공원역 3번출구  2020   8   1   0.833333   0.700000  0.992801   
506642  송파구  올림픽공원역 3번출구  2020   8   2   0.200000   0.200000  0.947368   
506643  송파구  올림픽공원역 3번출구  2020   8   3   0.166667   0.266667  0.624950   
506644  송파구  올림픽공원역 3번출구  2020   8   4   0.033333   0.033333  0.947368   
506645  송파구 

In [5]:
data = pd.read_csv('./올림픽공원역 3번출구 균형, 불균형.csv', encoding = 'cp949')
data

자치구         대여소명     년   월  시간        재고량  3사분위수  2사분위수  1사분위수   균형
0    송파구  올림픽공원역 3번출구  2020   6   0  -1.633333    6.2   4.05    2.6  불균형
1    송파구  올림픽공원역 3번출구  2020   6   1  -1.500000    6.2   4.05    2.6  불균형
2    송파구  올림픽공원역 3번출구  2020   6   2  -1.400000    6.2   4.05    2.6  불균형
3    송파구  올림픽공원역 3번출구  2020   6   3  -1.133333    6.2   4.05    2.6  불균형
4    송파구  올림픽공원역 3번출구  2020   6   4  -1.000000    6.2   4.05    2.6  불균형
5    송파구  올림픽공원역 3번출구  2020   6   5  -0.566667    6.2   4.05    2.6  불균형
6    송파구  올림픽공원역 3번출구  2020   6   6   1.433333    6.2   4.05    2.6  불균형
7    송파구  올림픽공원역 3번출구  2020   6   7   2.066667    6.2   4.05    2.6  불균형
8    송파구  올림픽공원역 3번출구  2020   6   8   5.100000    6.2   4.05    2.6   균형
9    송파구  올림픽공원역 3번출구  2020   6   9   5.300000    6.2   4.05    2.6   균형
10   송파구  올림픽공원역 3번출구  2020   6  10   4.433333    6.2   4.05    2.6   균형
11   송파구  올림픽공원역 3번출구  2020   6  11   4.100000    6.2   4.05    2.6   균형
12   송파구  올림픽공원역 3번출구  2020   6  12   5.100000    6.2   4.05    2.6   균형
13   송파구  올림픽공원역 3번출구  2020   6  13   3.933333    6.2   4.05    2.6   균형
14   송파구  올림픽공원역 3번출구  2020   6  14   3.700000    6.2   4.05    2.6   균형
15   송파구  올림픽공원역 3번출구  2020   6  15   2.600000    6.2   4.05    2.6   균형
16   송파구  올림픽공원역 3번출구  2020   6  16   2.400000    6.2   4.05    2.6  불균형
17   송파구  올림픽공원역 3번출구  2020   6  17   0.800000    6.2   4.05    2.6  불균형
18   송파구  올림픽공원역 3번출구  2020   6  18  -0.166667    6.2   4.05    2.6  불균형
19   송파구  올림픽공원역 3번출구  2020   6  19  -0.966667    6.2   4.05    2.6  불균형
20   송파구  올림픽공원역 3번출구  2020   6  20  -0.366667    6.2   4.05    2.6  불균형
21   송파구  올림픽공원역 3번출구  2020   6  21  -1.400000    6.2   4.05    2.6  불균형
22   송파구  올림픽공원역 3번출구  2020   6  22  -1.833333    6.2   4.05    2.6  불균형
23   송파구  올림픽공원역 3번출구  2020   6  23  -2.133333    6.2   4.05    2.6  불균형
24   송파구  올림픽공원역 3번출구  2020   7   0  -0.100000    6.2   4.05    2.6  불균형
25   송파구  올림픽공원역 3번출구  2020   7   1   0.233333    6.2   4.05    2.6  불균형
26   송파구  올림픽공원역 3번출구  2020   7   2   0.366667    6.2   4.05    2.6  불균형
27   송파구  올림픽공원역 3번출구  2020   7   3   0.400000    6.2   4.05    2.6  불균형
28   송파구  올림픽공원역 3번출구  2020   7   4   0.433333    6.2   4.05    2.6  불균형
29   송파구  올림픽공원역 3번출구  2020   7   5   0.766667    6.2   4.05    2.6  불균형
30   송파구  올림픽공원역 3번출구  2020   7   6   2.900000    6.2   4.05    2.6   균형
31   송파구  올림픽공원역 3번출구  2020   7   7   3.633333    6.2   4.05    2.6   균형
32   송파구  올림픽공원역 3번출구  2020   7   8   4.766667    6.2   4.05    2.6   균형
33   송파구  올림픽공원역 3번출구  2020   7   9   3.966667    6.2   4.05    2.6   균형
34   송파구  올림픽공원역 3번출구  2020   7  10   3.766667    6.2   4.05    2.6   균형
35   송파구  올림픽공원역 3번출구  2020   7  11   3.766667    6.2   4.05    2.6   균형
36   송파구  올림픽공원역 3번출구  2020   7  12   3.866667    6.2   4.05    2.6   균형
37   송파구  올림픽공원역 3번출구  2020   7  13   3.866667    6.2   4.05    2.6   균형
38   송파구  올림픽공원역 3번출구  2020   7  14   3.100000    6.2   4.05    2.6   균형
39   송파구  올림픽공원역 3번출구  2020   7  15   2.100000    6.2   4.05    2.6  불균형
40   송파구  올림픽공원역 3번출구  2020   7  16   3.000000    6.2   4.05    2.6   균형
41   송파구  올림픽공원역 3번출구  2020   7  17   1.500000    6.2   4.05    2.6  불균형
42   송파구  올림픽공원역 3번출구  2020   7  18   1.233333    6.2   4.05    2.6  불균형
43   송파구  올림픽공원역 3번출구  2020   7  19   0.733333    6.2   4.05    2.6  불균형
44   송파구  올림픽공원역 3번출구  2020   7  20   1.100000    6.2   4.05    2.6  불균형
45   송파구  올림픽공원역 3번출구  2020   7  21   0.566667    6.2   4.05    2.6  불균형
46   송파구  올림픽공원역 3번출구  2020   7  22  -0.366667    6.2   4.05    2.6  불균형
47   송파구  올림픽공원역 3번출구  2020   7  23  -0.300000    6.2   4.05    2.6  불균형
48   송파구  올림픽공원역 3번출구  2020   8   0   4.266667    6.2   4.05    2.6   균형
49   송파구  올림픽공원역 3번출구  2020   8   1   4.466667    6.2   4.05    2.6   균형
50   송파구  올림픽공원역 3번출구  2020   8   2   4.533333    6.2   4.05    2.6   균형
51   송파구  올림픽공원역 3번출구  2020   8   3   4.600000    6.2   4.05    2.6   균형
52   송파구  올림픽공원역 3번출구  2020   8   4   4.633333    6.2   4.05    2.6   균형
53   송파구  올림픽공원역 3번출구  2020   8   5   5.033333    6.2   4.05   

In [6]:
df_new = pd.merge(df,data, on = ['자치구', '대여소명','년', '월', '시간'],how='outer')
df_new

자치구         대여소명     년   월  시간     lambda         mu      p_1+  \
0    송파구  올림픽공원역 3번출구  2020   6   0   2.500000   2.000000  0.996344   
1    송파구  올림픽공원역 3번출구  2020   6   1   1.033333   1.166667  0.873062   
2    송파구  올림픽공원역 3번출구  2020   6   2   0.500000   0.533333  0.911549   
3    송파구  올림픽공원역 3번출구  2020   6   3   0.433333   0.333333  0.997934   
4    송파구  올림픽공원역 3번출구  2020   6   4   0.133333   0.100000  0.998585   
5    송파구  올림픽공원역 3번출구  2020   6   5   0.733333   0.333333  1.000000   
6    송파구  올림픽공원역 3번출구  2020   6   6   3.866667   1.900000  0.999999   
7    송파구  올림픽공원역 3번출구  2020   6   7   4.366667   3.733333  0.990896   
8    송파구  올림픽공원역 3번출구  2020   6   8   7.000000   6.900000  0.953905   
9    송파구  올림픽공원역 3번출구  2020   6   9   6.066667   5.866667  0.961724   
10   송파구  올림픽공원역 3번출구  2020   6  10   5.033333   5.900000  0.845559   
11   송파구  올림픽공원역 3번출구  2020   6  11   5.933333   6.266667  0.917663   
12   송파구  올림픽공원역 3번출구  2020   6  12   7.400000   6.466667  0.987928   
13   송파구  올림픽공원역 3번출구  2020   6  13   6.766667   8.033333  0.836031   
14   송파구  올림픽공원역 3번출구  2020   6  14   6.366667   6.600000  0.928628   
15   송파구  올림픽공원역 3번출구  2020   6  15   7.633333   8.733333  0.863468   
16   송파구  올림픽공원역 3번출구  2020   6  16  11.266667  11.500000  0.937102   
17   송파구  올림픽공원역 3번출구  2020   6  17  14.600000  16.200000  0.885337   
18   송파구  올림픽공원역 3번출구  2020   6  18  16.366667  17.366667  0.914812   
19   송파구  올림픽공원역 3번출구  2020   6  19  17.600000  18.533333  0.919470   
20   송파구  올림픽공원역 3번출구  2020   6  20  21.900000  21.333333  0.958856   
21   송파구  올림픽공원역 3번출구  2020   6  21  17.233333  18.333333  0.913217   
22   송파구  올림픽공원역 3번출구  2020   6  22  12.533333  12.966667  0.929757   
23   송파구  올림픽공원역 3번출구  2020   6  23   4.633333   5.000000  0.904107   
24   송파구  올림픽공원역 3번출구  2020   7   0   1.400000   1.300000  0.975090   
25   송파구  올림픽공원역 3번출구  2020   7   1   0.700000   0.700000  0.947368   
26   송파구  올림픽공원역 3번출구  2020   7   2   0.433333   0.333333  0.997934   
27   송파구  올림픽공원역 3번출구  2020   7   3   0.266667   0.466667  0.571418   
28   송파구  올림픽공원역 3번출구  2020   7   4   0.166667   0.166667  0.947368   
29   송파구  올림픽공원역 3번출구  2020   7   5   0.466667   0.266667  0.999982   
30   송파구  올림픽공원역 3번출구  2020   7   6   3.333333   1.200000  1.000000   
31   송파구  올림픽공원역 3번출구  2020   7   7   4.166667   3.533333  0.991828   
32   송파구  올림픽공원역 3번출구  2020   7   8   4.933333   4.933333  0.947368   
33   송파구  올림픽공원역 3번출구  2020   7   9   4.600000   5.433333  0.839855   
34   송파구  올림픽공원역 3번출구  2020   7  10   4.100000   4.300000  0.921886   
35   송파구  올림픽공원역 3번출구  2020   7  11   4.400000   4.500000  0.936056   
36   송파구  올림픽공원역 3번출구  2020   7  12   5.966667   6.033333  0.941941   
37   송파구  올림픽공원역 3번출구  2020   7  13   6.666667   6.666667  0.947368   
38   송파구  올림픽공원역 3번출구  2020   7  14   7.300000   8.100000  0.885337   
39   송파구  올림픽공원역 3번출구  2020   7  15   8.300000   9.300000  0.878479   
40   송파구  올림픽공원역 3번출구  2020   7  16   9.566667   8.666667  0.981239   
41   송파구  올림픽공원역 3번출구  2020   7  17  10.233333  11.766667  0.859811   
42   송파구  올림픽공원역 3번출구  2020   7  18  12.766667  13.100000  0.934284   
43   송파구  올림픽공원역 3번출구  2020   7  19  13.266667  13.833333  0.925287   
44   송파구  올림픽공원역 3번출구  2020   7  20  15.933333  15.700000  0.954063   
45   송파구  올림픽공원역 3번출구  2020   7  21  14.200000  14.800000  0.925543   
46   송파구  올림픽공원역 3번출구  2020   7  22   8.733333   9.800000  0.877432   
47   송파구  올림픽공원역 3번출구  2020   7  23   5.133333   5.100000  0.950397   
48   송파구  올림픽공원역 3번출구  2020   8   0   0.966667   0.900000  0.974345   
49   송파구  올림픽공원역 3번출구  2020   8   1   0.833333   0.700000  0.992801   
50   송파구  올림픽공원역 3번출구  2020   8   2   0.200000   0.200000  0.947368   
51   송파구  올림픽공원역 3번출구  2020   8   3   0.166667   0.266667  0.624950   
52   송파구  올림픽공원역 3번출구  2020   8   4   0.033333   0.033333  0.947368   
53   송파구  올림픽공원역 3번출구  2020   8   5   0.400000   0.066667  1.000000   
54   송파구  올림픽공원역 3번출구  2020   8   6   1.866667   1.033333  0.999989   
55   송파구  올림픽공원역 3번출구  2020  

In [7]:
df_new['내 균형'] = '균형'
df_new

자치구         대여소명     년   월  시간     lambda         mu      p_1+  \
0    송파구  올림픽공원역 3번출구  2020   6   0   2.500000   2.000000  0.996344   
1    송파구  올림픽공원역 3번출구  2020   6   1   1.033333   1.166667  0.873062   
2    송파구  올림픽공원역 3번출구  2020   6   2   0.500000   0.533333  0.911549   
3    송파구  올림픽공원역 3번출구  2020   6   3   0.433333   0.333333  0.997934   
4    송파구  올림픽공원역 3번출구  2020   6   4   0.133333   0.100000  0.998585   
5    송파구  올림픽공원역 3번출구  2020   6   5   0.733333   0.333333  1.000000   
6    송파구  올림픽공원역 3번출구  2020   6   6   3.866667   1.900000  0.999999   
7    송파구  올림픽공원역 3번출구  2020   6   7   4.366667   3.733333  0.990896   
8    송파구  올림픽공원역 3번출구  2020   6   8   7.000000   6.900000  0.953905   
9    송파구  올림픽공원역 3번출구  2020   6   9   6.066667   5.866667  0.961724   
10   송파구  올림픽공원역 3번출구  2020   6  10   5.033333   5.900000  0.845559   
11   송파구  올림픽공원역 3번출구  2020   6  11   5.933333   6.266667  0.917663   
12   송파구  올림픽공원역 3번출구  2020   6  12   7.400000   6.466667  0.987928   
13   송파구  올림픽공원역 3번출구  2020   6  13   6.766667   8.033333  0.836031   
14   송파구  올림픽공원역 3번출구  2020   6  14   6.366667   6.600000  0.928628   
15   송파구  올림픽공원역 3번출구  2020   6  15   7.633333   8.733333  0.863468   
16   송파구  올림픽공원역 3번출구  2020   6  16  11.266667  11.500000  0.937102   
17   송파구  올림픽공원역 3번출구  2020   6  17  14.600000  16.200000  0.885337   
18   송파구  올림픽공원역 3번출구  2020   6  18  16.366667  17.366667  0.914812   
19   송파구  올림픽공원역 3번출구  2020   6  19  17.600000  18.533333  0.919470   
20   송파구  올림픽공원역 3번출구  2020   6  20  21.900000  21.333333  0.958856   
21   송파구  올림픽공원역 3번출구  2020   6  21  17.233333  18.333333  0.913217   
22   송파구  올림픽공원역 3번출구  2020   6  22  12.533333  12.966667  0.929757   
23   송파구  올림픽공원역 3번출구  2020   6  23   4.633333   5.000000  0.904107   
24   송파구  올림픽공원역 3번출구  2020   7   0   1.400000   1.300000  0.975090   
25   송파구  올림픽공원역 3번출구  2020   7   1   0.700000   0.700000  0.947368   
26   송파구  올림픽공원역 3번출구  2020   7   2   0.433333   0.333333  0.997934   
27   송파구  올림픽공원역 3번출구  2020   7   3   0.266667   0.466667  0.571418   
28   송파구  올림픽공원역 3번출구  2020   7   4   0.166667   0.166667  0.947368   
29   송파구  올림픽공원역 3번출구  2020   7   5   0.466667   0.266667  0.999982   
30   송파구  올림픽공원역 3번출구  2020   7   6   3.333333   1.200000  1.000000   
31   송파구  올림픽공원역 3번출구  2020   7   7   4.166667   3.533333  0.991828   
32   송파구  올림픽공원역 3번출구  2020   7   8   4.933333   4.933333  0.947368   
33   송파구  올림픽공원역 3번출구  2020   7   9   4.600000   5.433333  0.839855   
34   송파구  올림픽공원역 3번출구  2020   7  10   4.100000   4.300000  0.921886   
35   송파구  올림픽공원역 3번출구  2020   7  11   4.400000   4.500000  0.936056   
36   송파구  올림픽공원역 3번출구  2020   7  12   5.966667   6.033333  0.941941   
37   송파구  올림픽공원역 3번출구  2020   7  13   6.666667   6.666667  0.947368   
38   송파구  올림픽공원역 3번출구  2020   7  14   7.300000   8.100000  0.885337   
39   송파구  올림픽공원역 3번출구  2020   7  15   8.300000   9.300000  0.878479   
40   송파구  올림픽공원역 3번출구  2020   7  16   9.566667   8.666667  0.981239   
41   송파구  올림픽공원역 3번출구  2020   7  17  10.233333  11.766667  0.859811   
42   송파구  올림픽공원역 3번출구  2020   7  18  12.766667  13.100000  0.934284   
43   송파구  올림픽공원역 3번출구  2020   7  19  13.266667  13.833333  0.925287   
44   송파구  올림픽공원역 3번출구  2020   7  20  15.933333  15.700000  0.954063   
45   송파구  올림픽공원역 3번출구  2020   7  21  14.200000  14.800000  0.925543   
46   송파구  올림픽공원역 3번출구  2020   7  22   8.733333   9.800000  0.877432   
47   송파구  올림픽공원역 3번출구  2020   7  23   5.133333   5.100000  0.950397   
48   송파구  올림픽공원역 3번출구  2020   8   0   0.966667   0.900000  0.974345   
49   송파구  올림픽공원역 3번출구  2020   8   1   0.833333   0.700000  0.992801   
50   송파구  올림픽공원역 3번출구  2020   8   2   0.200000   0.200000  0.947368   
51   송파구  올림픽공원역 3번출구  2020   8   3   0.166667   0.266667  0.624950   
52   송파구  올림픽공원역 3번출구  2020   8   4   0.033333   0.033333  0.947368   
53   송파구  올림픽공원역 3번출구  2020   8   5   0.400000   0.066667  1.000000   
54   송파구  올림픽공원역 3번출구  2020   8   6   1.866667   1.033333  0.999989   
55   송파구  올림픽공원역 3번출구  2020  

# 일단 L_s, 기존균형 비교

## L_s가 Q1보다 작거나, Q3보다 크면 불균형

### L_s가 q1보다 작지만 이때 p_1+이 충분히 크다면 균형
- ls가 q1보다 작고 p_1+가 0.5보다 작으면 불균형

In [8]:
# ls가 q1보다 작으므로 불균형

e = df_new[(df_new['L_s'] < df_new['1사분위수']) & (df_new['p_1+'] < 0.6)]
e

자치구         대여소명     년   월  시간    lambda        mu      p_1+       L_s  \
27   송파구  올림픽공원역 3번출구  2020   7   3  0.266667  0.466667  0.571418  1.332875   
98   송파구  올림픽공원역 3번출구  2020  10   2  0.200000  0.466667  0.428571  0.749998   
124  송파구  올림픽공원역 3번출구  2020  11   4  0.000000  0.133333  0.000000  0.000000   
125  송파구  올림픽공원역 3번출구  2020  11   5  0.066667  0.133333  0.499999  0.999964   
129  송파구  올림픽공원역 3번출구  2020  11   9  2.866667  5.233333  0.547766  1.211062   
167  송파구  올림픽공원역 3번출구  2020  12  23  0.366667  0.833333  0.440000  0.785711   
169  송파구  올림픽공원역 3번출구  2021   1   1  0.033333  0.133333  0.250000  0.333333   
170  송파구  올림픽공원역 3번출구  2021   1   2  0.033333  0.166667  0.200000  0.250000   
171  송파구  올림픽공원역 3번출구  2021   1   3  0.033333  0.066667  0.499999  0.999964   
172  송파구  올림픽공원역 3번출구  2021   1   4  0.000000  0.033333  0.000000  0.000000   
191  송파구  올림픽공원역 3번출구  2021   1  23  0.266667  0.533333  0.499999  0.999964   
194  송파구  올림픽공원역 3번출구  2021   2   2  0.100000  0.000000  0.000000  0.000000   
196  송파구  올림픽공원역 3번출구  2021   2   4  0.033333  0.100000  0.333333  0.500000   
219  송파구  올림픽공원역 3번출구  2021   3   3  0.166667  0.000000  0.000000  0.000000   
241  송파구  올림픽공원역 3번출구  2021   4   1  0.133333  0.433333  0.307692  0.444444   
242  송파구  올림픽공원역 3번출구  2021   4   2  0.066667  0.133333  0.499999  0.999964   

           W_s       재고량  3사분위수  2사분위수  1사분위수   균형 내 균형  
27    4.998372  0.400000    6.2   4.05    2.6  불균형   균형  
98    3.749991  3.033333    6.2   4.05    2.6   균형   균형  
124   0.000000  5.333333    6.2   4.05    2.6   균형   균형  
125  14.999485  5.333333    6.2   4.05    2.6   균형   균형  
129   0.422467  6.666667    6.2   4.05    2.6  불균형   균형  
167   2.142849  1.566667    6.2   4.05    2.6  불균형   균형  
169  10.000000  3.900000    6.2   4.05    2.6   균형   균형  
170   7.500000  3.900000    6.2   4.05    2.6   균형   균형  
171  29.998970  3.933333    6.2   4.05    2.6   균형   균형  
172   0.000000  3.933333    6.2   4.05    2.6   균형   균형  
191   3.749871  4.666667    6.2   4.05    2.6   균형   균형  
194   0.000000  6.166667    6.2   4.05    2.6   균형   균형  
196  15.000000  6.333333    6.2   4.05    2.6  불균형   균형  
219   0.000000  4.166667    6.2   4.05    2.6   균형   균형  
241   3.333333  2.900000    6.2   4.05    2.6   균형   균형  
242  14.999485  2.866667    6.2   4.05    2.6   균형   균형

In [9]:
for i in range(len(df_new)):
    if (df_new['L_s'][i] < df_new['1사분위수'][i]) & (df_new['p_1+'][i] < 0.6):
        df_new['내 균형'][i] = '불균형'

In [10]:
# ls가 q3보다 크므로 불균형

r = df_new[(df_new['L_s'] > df_new['3사분위수']) & (df_new['L_s'] > (20*0.7))]
r

자치구         대여소명     년   월  시간     lambda        mu      p_1+        L_s  \
0    송파구  올림픽공원역 3번출구  2020   6   0   2.500000  2.000000  0.996344  14.277823   
3    송파구  올림픽공원역 3번출구  2020   6   3   0.433333  0.333333  0.997934  14.797528   
4    송파구  올림픽공원역 3번출구  2020   6   4   0.133333  0.100000  0.998585  15.080679   
5    송파구  올림픽공원역 3번출구  2020   6   5   0.733333  0.333333  1.000000  17.166673   
6    송파구  올림픽공원역 3번출구  2020   6   6   3.866667  1.900000  0.999999  17.033924   
26   송파구  올림픽공원역 3번출구  2020   7   2   0.433333  0.333333  0.997934  14.797528   
29   송파구  올림픽공원역 3번출구  2020   7   5   0.466667  0.266667  0.999982  16.667125   
30   송파구  올림픽공원역 3번출구  2020   7   6   3.333333  1.200000  1.000000  17.437500   
53   송파구  올림픽공원역 3번출구  2020   8   5   0.400000  0.066667  1.000000  17.800000   
54   송파구  올림픽공원역 3번출구  2020   8   6   1.866667  1.033333  0.999989  16.760251   
61   송파구  올림픽공원역 3번출구  2020   8  13   3.166667  2.400000  0.998343  14.968095   
64   송파구  올림픽공원역 3번출구  2020   8  16   5.533333  4.466667  0.995846  14.143007   
73   송파구  올림픽공원역 3번출구  2020   9   1   1.066667  0.866667  0.995447  14.041539   
74   송파구  올림픽공원역 3번출구  2020   9   2   0.366667  0.133333  1.000000  17.428572   
76   송파구  올림픽공원역 3번출구  2020   9   4   0.333333  0.166667  0.999998  17.000036   
77   송파구  올림픽공원역 3번출구  2020   9   5   0.300000  0.200000  0.999774  16.008575   
78   송파구  올림픽공원역 3번출구  2020   9   6   3.666667  1.633333  1.000000  17.196725   
96   송파구  올림픽공원역 3번출구  2020  10   0   1.466667  0.833333  0.999984  16.684622   
97   송파구  올림픽공원역 3번출구  2020  10   1   1.100000  0.766667  0.999543  15.719965   
100  송파구  올림픽공원역 3번출구  2020  10   4   0.133333  0.066667  0.999998  17.000036   
101  송파구  올림픽공원역 3번출구  2020  10   5   0.300000  0.233333  0.997569  14.661692   
102  송파구  올림픽공원역 3번출구  2020  10   6   2.966667  0.833333  1.000000  17.609375   
103  송파구  올림픽공원역 3번출구  2020  10   7   5.366667  4.200000  0.997338  14.582072   
120  송파구  올림픽공원역 3번출구  2020  11   0   0.533333  0.233333  1.000000  17.222225   
121  송파구  올림픽공원역 3번출구  2020  11   1   0.500000  0.366667  0.998994  15.302563   
123  송파구  올림픽공원역 3번출구  2020  11   3   0.133333  0.066667  0.999998  17.000036   
126  송파구  올림픽공원역 3번출구  2020  11   6   2.000000  0.266667  1.000000  17.846154   
127  송파구  올림픽공원역 3번출구  2020  11   7   3.933333  2.566667  0.999840  16.127659   
144  송파구  올림픽공원역 3번출구  2020  12   0   0.433333  0.266667  0.999938  16.401873   
145  송파구  올림픽공원역 3번출구  2020  12   1   0.266667  0.166667  0.999921  16.335848   
146  송파구  올림픽공원역 3번출구  2020  12   2   0.166667  0.066667  1.000000  17.333334   
149  송파구  올림픽공원역 3번출구  2020  12   5   0.100000  0.033333  1.000000  17.500000   
150  송파구  올림픽공원역 3번출구  2020  12   6   1.166667  0.200000  1.000000  17.793103   
151  송파구  올림픽공원역 3번출구  2020  12   7   1.366667  0.900000  0.999815  16.078219   
152  송파구  올림픽공원역 3번출구  2020  12   8   3.766667  2.900000  0.997907  14.786941   
155  송파구  올림픽공원역 3번출구  2020  12  11   2.133333  1.500000  0.999476  15.655179   
168  송파구  올림픽공원역 3번출구  2021   1   0   0.166667  0.133333  0.996344  14.277823   
174  송파구  올림픽공원역 3번출구  2021   1   6   0.933333  0.166667  1.000000  17.782609   
175  송파구  올림픽공원역 3번출구  2021   1   7   1.366667  0.466667  1.000000  17.481482   
177  송파구  올림픽공원역 3번출구  2021   1   9   1.166667  0.933333  0.996344  14.277823   
184  송파구  올림픽공원역 3번출구  2021   1  16   6.066667  4.700000  0.997705  14.710935   
193  송파구  올림픽공원역 3번출구  2021   2   1   0.100000  0.033333  1.000000  17.500000   
195  송파구  올림픽공원역 3번출구  2021   2   3   0.166667  0.033333  1.000000  17.750000   
197  송파구  올림픽공원역 3번출구  2021   2   5   0.366667  0.100000  1.000000  17.625000   
198  송파구  올림픽공원역 3번출구  2021   2   6   1.333333  0.800000  0.999959  16.501158   
199  송파구  올림픽공원역 3번출구  2021   2   7   0.600000  0.333333  0.999989  16.750268   
201  송파구  올림픽공원역 3번출구  2021   2   9   2.566667  1.633333  0.999893  16.253542   
220  송파구  올림픽공원역 3번출구  2021   3   4   0.133333  0.033333  1.000000  17.666667   
221  송파구  올림픽공원역 3번출구  2021   3   5 

In [11]:
for i in range(len(df_new)):
    if (df_new['L_s'][i] > df_new['3사분위수'][i]) & (df_new['L_s'][i] > (20*0.7)):
        df_new['내 균형'][i] = '불균형'

# L_s가 Q1~Q3사이이더라도 이때 p_1+가 너무 작으면(50%이하) 불균형
- 따릉이 목표 거치율이 60~70%이므로 한대라도 있을 확률인 p_1+도 유사하게 잡기 위해

In [12]:
t = df_new[(df_new['3사분위수']>=df_new['L_s']) & (df_new['L_s'] >= df_new['1사분위수'])]
t

자치구         대여소명     년   월  시간     lambda         mu      p_1+       L_s  \
1    송파구  올림픽공원역 3번출구  2020   6   1   1.033333   1.166667  0.873062  5.646547   
10   송파구  올림픽공원역 3번출구  2020   6  10   5.033333   5.900000  0.845559  4.831388   
13   송파구  올림픽공원역 3번출구  2020   6  13   6.766667   8.033333  0.836031  4.583788   
15   송파구  올림픽공원역 3번출구  2020   6  15   7.633333   8.733333  0.863468  5.343794   
17   송파구  올림픽공원역 3번출구  2020   6  17  14.600000  16.200000  0.885337  6.066611   
33   송파구  올림픽공원역 3번출구  2020   7   9   4.600000   5.433333  0.839855  4.681210   
38   송파구  올림픽공원역 3번출구  2020   7  14   7.300000   8.100000  0.885337  6.066611   
39   송파구  올림픽공원역 3번출구  2020   7  15   8.300000   9.300000  0.878479  5.827152   
41   송파구  올림픽공원역 3번출구  2020   7  17  10.233333  11.766667  0.859811  5.233813   
46   송파구  올림픽공원역 3번출구  2020   7  22   8.733333   9.800000  0.877432  5.791680   
55   송파구  올림픽공원역 3번출구  2020   8   7   1.900000   2.166667  0.865861  5.417348   
57   송파구  올림픽공원역 3번출구  2020   8   9   2.533333   3.033333  0.829604  4.425701   
62   송파구  올림픽공원역 3번출구  2020   8  14   3.200000   3.700000  0.855719  5.114088   
63   송파구  올림픽공원역 3번출구  2020   8  15   3.033333   3.600000  0.836270  4.589825   
65   송파구  올림픽공원역 3번출구  2020   8  17   5.733333   7.066667  0.807701  3.935527   
75   송파구  올림픽공원역 3번출구  2020   9   3   0.200000   0.266667  0.748938  2.919321   
81   송파구  올림픽공원역 3번출구  2020   9   9   3.600000   4.833333  0.743878  2.848218   
82   송파구  올림픽공원역 3번출구  2020   9  10   4.500000   5.000000  0.884382  6.032519   
99   송파구  올림픽공원역 3번출구  2020  10   3   0.166667   0.200000  0.827948  4.386067   
105  송파구  올림픽공원역 3번출구  2020  10   9   5.500000   6.933333  0.790700  3.601102   
107  송파구  올림픽공원역 3번출구  2020  10  11   7.466667   8.400000  0.875619  5.730914   
118  송파구  올림픽공원역 3번출구  2020  10  22   7.200000   8.600000  0.831447  4.470334   
130  송파구  올림픽공원역 3번출구  2020  11  10   3.000000   3.400000  0.870329  5.558131   
134  송파구  올림픽공원역 3번출구  2020  11  14   7.033333   7.900000  0.876745  5.768556   
139  송파구  올림픽공원역 3번출구  2020  11  19   4.200000   5.266667  0.794682  3.676138   
141  송파구  올림픽공원역 3번출구  2020  11  21   4.400000   5.366667  0.815641  4.104979   
153  송파구  올림픽공원역 3번출구  2020  12   9   2.433333   3.066667  0.790899  3.604799   
156  송파구  올림픽공원역 3번출구  2020  12  12   3.000000   3.333333  0.884382  6.032519   
162  송파구  올림픽공원역 3번출구  2020  12  18   4.200000   5.000000  0.833953  4.531923   
165  송파구  올림픽공원역 3번출구  2020  12  21   2.733333   3.166667  0.854260  5.072235   
180  송파구  올림픽공원역 3번출구  2021   1  12   1.900000   2.233333  0.843489  4.776211   
188  송파구  올림픽공원역 3번출구  2021   1  20   1.866667   2.066667  0.886868  6.121830   
192  송파구  올림픽공원역 3번출구  2021   2   0   0.300000   0.400000  0.748938  2.919321   
202  송파구  올림픽공원역 3번출구  2021   2  10   2.566667   2.866667  0.880750  5.905102   
203  송파구  올림픽공원역 3번출구  2021   2  11   3.300000   3.700000  0.878017  5.811476   
205  송파구  올림픽공원역 3번출구  2021   2  13   6.500000   7.633333  0.844176  4.794429   
206  송파구  올림픽공원역 3번출구  2021   2  14   6.666667   8.000000  0.827948  4.386067   
213  송파구  올림픽공원역 3번출구  2021   2  21   2.666667   3.400000  0.782159  3.446528   
214  송파구  올림픽공원역 3번출구  2021   2  22   2.166667   2.966667  0.729647  2.659714   
215  송파구  올림픽공원역 3번출구  2021   2  23   0.766667   1.000000  0.765159  3.162949   
216  송파구  올림픽공원역 3번출구  2021   3   0   0.666667   0.833333  0.797076  3.722177   
217  송파구  올림픽공원역 3번출구  2021   3   1   0.200000   0.266667  0.748938  2.919321   
218  송파구  올림픽공원역 3번출구  2021   3   2   0.100000   0.133333  0.748938  2.919321   
227  송파구  올림픽공원역 3번출구  2021   3  11   4.733333   5.733333  0.820887  4.221975   
229  송파구  올림픽공원역 3번출구  2021   3  13   6.400000   7.500000  0.845757  4.836698   
230  송파구  올림픽공원역 3번출구  2021   3  14   7.866667   9.900000  0.791976  3.624932   
235  송파구  올림픽공원역 3번출구  2021   3  19   9.333333  11.133333  0.832449  4.494829   
237  송파구  올림픽공원역 3번출구  2021   3  21   7.066667   8.166667  0.856095  5.124951   
250  송파구  올림픽공원역 3번출구  2021   4  10 

In [13]:
t[t['p_1+']< 0.6]

Empty DataFrame
Columns: [자치구, 대여소명, 년, 월, 시간, lambda, mu, p_1+, L_s, W_s, 재고량, 3사분위수, 2사분위수, 1사분위수, 균형, 내 균형]
Index: []

- 위 결과를 보면 l_s가 q1~q3사이에 있지만 자전거가 한대라도 있을 확률이 0.5이하이므로 불균형이다
- 특히 19시, 20시의 경우 퇴근시간이므로 이시간에 불균형이면 이용에 불편을 초래함
- 또한 21년 1월 19시의 경우 불균형인데도 불구하고 기존연구방법에서는 이를 균형으로 판단

In [14]:
for i in range(len(df_new)):
    if (df_new['3사분위수'][i]>=df_new['L_s'][i]) & (df_new['L_s'][i] >= df_new['1사분위수'][i]) & (df_new['p_1+'][i] < 0.6):
        df_new['내 균형'][i] = '불균형'

In [15]:
df_new

자치구         대여소명     년   월  시간     lambda         mu      p_1+  \
0    송파구  올림픽공원역 3번출구  2020   6   0   2.500000   2.000000  0.996344   
1    송파구  올림픽공원역 3번출구  2020   6   1   1.033333   1.166667  0.873062   
2    송파구  올림픽공원역 3번출구  2020   6   2   0.500000   0.533333  0.911549   
3    송파구  올림픽공원역 3번출구  2020   6   3   0.433333   0.333333  0.997934   
4    송파구  올림픽공원역 3번출구  2020   6   4   0.133333   0.100000  0.998585   
5    송파구  올림픽공원역 3번출구  2020   6   5   0.733333   0.333333  1.000000   
6    송파구  올림픽공원역 3번출구  2020   6   6   3.866667   1.900000  0.999999   
7    송파구  올림픽공원역 3번출구  2020   6   7   4.366667   3.733333  0.990896   
8    송파구  올림픽공원역 3번출구  2020   6   8   7.000000   6.900000  0.953905   
9    송파구  올림픽공원역 3번출구  2020   6   9   6.066667   5.866667  0.961724   
10   송파구  올림픽공원역 3번출구  2020   6  10   5.033333   5.900000  0.845559   
11   송파구  올림픽공원역 3번출구  2020   6  11   5.933333   6.266667  0.917663   
12   송파구  올림픽공원역 3번출구  2020   6  12   7.400000   6.466667  0.987928   
13   송파구  올림픽공원역 3번출구  2020   6  13   6.766667   8.033333  0.836031   
14   송파구  올림픽공원역 3번출구  2020   6  14   6.366667   6.600000  0.928628   
15   송파구  올림픽공원역 3번출구  2020   6  15   7.633333   8.733333  0.863468   
16   송파구  올림픽공원역 3번출구  2020   6  16  11.266667  11.500000  0.937102   
17   송파구  올림픽공원역 3번출구  2020   6  17  14.600000  16.200000  0.885337   
18   송파구  올림픽공원역 3번출구  2020   6  18  16.366667  17.366667  0.914812   
19   송파구  올림픽공원역 3번출구  2020   6  19  17.600000  18.533333  0.919470   
20   송파구  올림픽공원역 3번출구  2020   6  20  21.900000  21.333333  0.958856   
21   송파구  올림픽공원역 3번출구  2020   6  21  17.233333  18.333333  0.913217   
22   송파구  올림픽공원역 3번출구  2020   6  22  12.533333  12.966667  0.929757   
23   송파구  올림픽공원역 3번출구  2020   6  23   4.633333   5.000000  0.904107   
24   송파구  올림픽공원역 3번출구  2020   7   0   1.400000   1.300000  0.975090   
25   송파구  올림픽공원역 3번출구  2020   7   1   0.700000   0.700000  0.947368   
26   송파구  올림픽공원역 3번출구  2020   7   2   0.433333   0.333333  0.997934   
27   송파구  올림픽공원역 3번출구  2020   7   3   0.266667   0.466667  0.571418   
28   송파구  올림픽공원역 3번출구  2020   7   4   0.166667   0.166667  0.947368   
29   송파구  올림픽공원역 3번출구  2020   7   5   0.466667   0.266667  0.999982   
30   송파구  올림픽공원역 3번출구  2020   7   6   3.333333   1.200000  1.000000   
31   송파구  올림픽공원역 3번출구  2020   7   7   4.166667   3.533333  0.991828   
32   송파구  올림픽공원역 3번출구  2020   7   8   4.933333   4.933333  0.947368   
33   송파구  올림픽공원역 3번출구  2020   7   9   4.600000   5.433333  0.839855   
34   송파구  올림픽공원역 3번출구  2020   7  10   4.100000   4.300000  0.921886   
35   송파구  올림픽공원역 3번출구  2020   7  11   4.400000   4.500000  0.936056   
36   송파구  올림픽공원역 3번출구  2020   7  12   5.966667   6.033333  0.941941   
37   송파구  올림픽공원역 3번출구  2020   7  13   6.666667   6.666667  0.947368   
38   송파구  올림픽공원역 3번출구  2020   7  14   7.300000   8.100000  0.885337   
39   송파구  올림픽공원역 3번출구  2020   7  15   8.300000   9.300000  0.878479   
40   송파구  올림픽공원역 3번출구  2020   7  16   9.566667   8.666667  0.981239   
41   송파구  올림픽공원역 3번출구  2020   7  17  10.233333  11.766667  0.859811   
42   송파구  올림픽공원역 3번출구  2020   7  18  12.766667  13.100000  0.934284   
43   송파구  올림픽공원역 3번출구  2020   7  19  13.266667  13.833333  0.925287   
44   송파구  올림픽공원역 3번출구  2020   7  20  15.933333  15.700000  0.954063   
45   송파구  올림픽공원역 3번출구  2020   7  21  14.200000  14.800000  0.925543   
46   송파구  올림픽공원역 3번출구  2020   7  22   8.733333   9.800000  0.877432   
47   송파구  올림픽공원역 3번출구  2020   7  23   5.133333   5.100000  0.950397   
48   송파구  올림픽공원역 3번출구  2020   8   0   0.966667   0.900000  0.974345   
49   송파구  올림픽공원역 3번출구  2020   8   1   0.833333   0.700000  0.992801   
50   송파구  올림픽공원역 3번출구  2020   8   2   0.200000   0.200000  0.947368   
51   송파구  올림픽공원역 3번출구  2020   8   3   0.166667   0.266667  0.624950   
52   송파구  올림픽공원역 3번출구  2020   8   4   0.033333   0.033333  0.947368   
53   송파구  올림픽공원역 3번출구  2020   8   5   0.400000   0.066667  1.000000   
54   송파구  올림픽공원역 3번출구  2020   8   6   1.866667   1.033333  0.999989   
55   송파구  올림픽공원역 3번출구  2020  

In [16]:
len(df_new[df_new['균형'] == '불균형'])

143

In [17]:
len(df_new[df_new['내 균형'] == '불균형'])

80

In [18]:
len(df_new)

288

In [19]:
# TN

len(df_new[(df_new['균형'] == '불균형') & (df_new['내 균형'] == '불균형')])

39

In [20]:
# TP

len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')])

104

In [21]:
# FN

len(df_new[(df_new['균형'] == '불균형') & (df_new['내 균형'] == '균형')])

104

In [22]:
# FP

len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '불균형')])

41

In [23]:
# 정확도(기존방법과 내 방법의 일치율)

(len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')]) + len(df_new[(df_new['균형'] == '불균형') & (df_new['내 균형'] == '불균형')])) / len(df_new)

0.4965277777777778

In [24]:
# 정밀도(기존 방법에서 균형으로 판단한 데이터 중 내가 균형이라고 판단한 비율)

len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')]) / (len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')]) + len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '불균형')]))

0.7172413793103448

In [25]:
# 재현율(내 방법에서 균형으로 판단한 것 중 기존 방법에서도 균형으로 판단한 비율)

len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')]) / (len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')]) + len(df_new[(df_new['균형'] == '불균형') & (df_new['내 균형'] == '균형')]))

0.5

In [26]:
a = df_new[(df_new['균형'] == '불균형') & (df_new['내 균형'] == '균형')]
a

자치구         대여소명     년   월  시간     lambda         mu      p_1+  \
1    송파구  올림픽공원역 3번출구  2020   6   1   1.033333   1.166667  0.873062   
2    송파구  올림픽공원역 3번출구  2020   6   2   0.500000   0.533333  0.911549   
7    송파구  올림픽공원역 3번출구  2020   6   7   4.366667   3.733333  0.990896   
16   송파구  올림픽공원역 3번출구  2020   6  16  11.266667  11.500000  0.937102   
17   송파구  올림픽공원역 3번출구  2020   6  17  14.600000  16.200000  0.885337   
18   송파구  올림픽공원역 3번출구  2020   6  18  16.366667  17.366667  0.914812   
19   송파구  올림픽공원역 3번출구  2020   6  19  17.600000  18.533333  0.919470   
20   송파구  올림픽공원역 3번출구  2020   6  20  21.900000  21.333333  0.958856   
21   송파구  올림픽공원역 3번출구  2020   6  21  17.233333  18.333333  0.913217   
22   송파구  올림픽공원역 3번출구  2020   6  22  12.533333  12.966667  0.929757   
23   송파구  올림픽공원역 3번출구  2020   6  23   4.633333   5.000000  0.904107   
24   송파구  올림픽공원역 3번출구  2020   7   0   1.400000   1.300000  0.975090   
25   송파구  올림픽공원역 3번출구  2020   7   1   0.700000   0.700000  0.947368   
28   송파구  올림픽공원역 3번출구  2020   7   4   0.166667   0.166667  0.947368   
39   송파구  올림픽공원역 3번출구  2020   7  15   8.300000   9.300000  0.878479   
41   송파구  올림픽공원역 3번출구  2020   7  17  10.233333  11.766667  0.859811   
42   송파구  올림픽공원역 3번출구  2020   7  18  12.766667  13.100000  0.934284   
43   송파구  올림픽공원역 3번출구  2020   7  19  13.266667  13.833333  0.925287   
44   송파구  올림픽공원역 3번출구  2020   7  20  15.933333  15.700000  0.954063   
45   송파구  올림픽공원역 3번출구  2020   7  21  14.200000  14.800000  0.925543   
46   송파구  올림픽공원역 3번출구  2020   7  22   8.733333   9.800000  0.877432   
47   송파구  올림픽공원역 3번출구  2020   7  23   5.133333   5.100000  0.950397   
57   송파구  올림픽공원역 3번출구  2020   8   9   2.533333   3.033333  0.829604   
72   송파구  올림픽공원역 3번출구  2020   9   0   1.866667   1.866667  0.947368   
75   송파구  올림픽공원역 3번출구  2020   9   3   0.200000   0.266667  0.748938   
89   송파구  올림픽공원역 3번출구  2020   9  17  19.866667  20.166667  0.939970   
90   송파구  올림픽공원역 3번출구  2020   9  18  19.966667  20.433333  0.935720   
91   송파구  올림픽공원역 3번출구  2020   9  19  17.966667  18.200000  0.941035   
93   송파구  올림픽공원역 3번출구  2020   9  21  16.533333  17.700000  0.909241   
94   송파구  올림픽공원역 3번출구  2020   9  22  10.633333  11.566667  0.898858   
95   송파구  올림픽공원역 3번출구  2020   9  23   4.866667   4.466667  0.978167   
113  송파구  올림픽공원역 3번출구  2020  10  17  21.400000  22.333333  0.924786   
115  송파구  올림픽공원역 3번출구  2020  10  19  11.633333  12.733333  0.894692   
117  송파구  올림픽공원역 3번출구  2020  10  21  12.033333  12.733333  0.916512   
118  송파구  올림픽공원역 3번출구  2020  10  22   7.200000   8.600000  0.831447   
119  송파구  올림픽공원역 3번출구  2020  10  23   3.133333   3.433333  0.893957   
128  송파구  올림픽공원역 3번출구  2020  11   8   6.666667   6.100000  0.978923   
130  송파구  올림픽공원역 3번출구  2020  11  10   3.000000   3.400000  0.870329   
131  송파구  올림픽공원역 3번출구  2020  11  11   4.133333   3.533333  0.990913   
132  송파구  올림픽공원역 3번출구  2020  11  12   6.433333   5.533333  0.990154   
133  송파구  올림픽공원역 3번출구  2020  11  13   6.133333   5.933333  0.961582   
134  송파구  올림픽공원역 3번출구  2020  11  14   7.033333   7.900000  0.876745   
135  송파구  올림픽공원역 3번출구  2020  11  15   8.200000   8.133333  0.951145   
136  송파구  올림픽공원역 3번출구  2020  11  16   9.666667   9.700000  0.945722   
137  송파구  올림픽공원역 3번출구  2020  11  17   8.166667   8.666667  0.914739   
138  송파구  올림픽공원역 3번출구  2020  11  18   6.900000   6.200000  0.982978   
139  송파구  올림픽공원역 3번출구  2020  11  19   4.200000   5.266667  0.794682   
140  송파구  올림픽공원역 3번출구  2020  11  20   6.100000   5.400000  0.985808   
141  송파구  올림픽공원역 3번출구  2020  11  21   4.400000   5.366667  0.815641   
147  송파구  올림픽공원역 3번출구  2020  12   3   0.066667   0.100000  0.666516   
148  송파구  올림픽공원역 3번출구  2020  12   4   0.066667   0.100000  0.666516   
153  송파구  올림픽공원역 3번출구  2020  12   9   2.433333   3.066667  0.790899   
154  송파구  올림픽공원역 3번출구  2020  12  10   1.500000   1.566667  0.924323   
156  송파구  올림픽공원역 3번출구  2020  12  12   3.000000   3.333333  0.884382   
157  송파구  올림픽공원역 3번출구  2020  12  13   3.333333   3.633333  0.897495   
158  송파구  올림픽공원역 3번출구  2020  

In [27]:
len(a)

104

In [28]:
len(a[a['p_1+'] > 0.9])

66

In [29]:
len(a[a['p_1+'] > 0.8])

94

In [30]:
len(a[a['p_1+'] > 0.7])

99

In [31]:
len(a[a['p_1+'] > 0.6])

104

- 기존방법이 불균형으로 판단한 데이터 중 본 연구에서 균형으로 판단한 데이터는 104개이지만
- 이 중 대여소에 적어도 자전거가 1대 있을 확률이 70%가 넘는 데이터가 70개로  95%이상의 경우에 이용자가 대여소에 도착했을 때 자전거를 빌릴 수 있는 확률이 70%이상이었다.
- 따라서 이러한 경우에는 자전거 재배치를 위한 대여소 내 재고의 균형/ 불균형을 판단할 경우에는 재배치를 할 필요가 없다고 판단하여 균형으로 판단할 수 있다.

In [32]:
b = df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '불균형')]
b

자치구         대여소명     년   월  시간    lambda        mu      p_1+        L_s  \
30   송파구  올림픽공원역 3번출구  2020   7   6  3.333333  1.200000  1.000000  17.437500   
53   송파구  올림픽공원역 3번출구  2020   8   5  0.400000  0.066667  1.000000  17.800000   
54   송파구  올림픽공원역 3번출구  2020   8   6  1.866667  1.033333  0.999989  16.760251   
61   송파구  올림픽공원역 3번출구  2020   8  13  3.166667  2.400000  0.998343  14.968095   
64   송파구  올림픽공원역 3번출구  2020   8  16  5.533333  4.466667  0.995846  14.143007   
78   송파구  올림픽공원역 3번출구  2020   9   6  3.666667  1.633333  1.000000  17.196725   
96   송파구  올림픽공원역 3번출구  2020  10   0  1.466667  0.833333  0.999984  16.684622   
97   송파구  올림픽공원역 3번출구  2020  10   1  1.100000  0.766667  0.999543  15.719965   
98   송파구  올림픽공원역 3번출구  2020  10   2  0.200000  0.466667  0.428571   0.749998   
100  송파구  올림픽공원역 3번출구  2020  10   4  0.133333  0.066667  0.999998  17.000036   
101  송파구  올림픽공원역 3번출구  2020  10   5  0.300000  0.233333  0.997569  14.661692   
102  송파구  올림픽공원역 3번출구  2020  10   6  2.966667  0.833333  1.000000  17.609375   
120  송파구  올림픽공원역 3번출구  2020  11   0  0.533333  0.233333  1.000000  17.222225   
121  송파구  올림픽공원역 3번출구  2020  11   1  0.500000  0.366667  0.998994  15.302563   
123  송파구  올림픽공원역 3번출구  2020  11   3  0.133333  0.066667  0.999998  17.000036   
124  송파구  올림픽공원역 3번출구  2020  11   4  0.000000  0.133333  0.000000   0.000000   
125  송파구  올림픽공원역 3번출구  2020  11   5  0.066667  0.133333  0.499999   0.999964   
150  송파구  올림픽공원역 3번출구  2020  12   6  1.166667  0.200000  1.000000  17.793103   
151  송파구  올림픽공원역 3번출구  2020  12   7  1.366667  0.900000  0.999815  16.078219   
168  송파구  올림픽공원역 3번출구  2021   1   0  0.166667  0.133333  0.996344  14.277823   
169  송파구  올림픽공원역 3번출구  2021   1   1  0.033333  0.133333  0.250000   0.333333   
170  송파구  올림픽공원역 3번출구  2021   1   2  0.033333  0.166667  0.200000   0.250000   
171  송파구  올림픽공원역 3번출구  2021   1   3  0.033333  0.066667  0.499999   0.999964   
172  송파구  올림픽공원역 3번출구  2021   1   4  0.000000  0.033333  0.000000   0.000000   
174  송파구  올림픽공원역 3번출구  2021   1   6  0.933333  0.166667  1.000000  17.782609   
175  송파구  올림픽공원역 3번출구  2021   1   7  1.366667  0.466667  1.000000  17.481482   
177  송파구  올림픽공원역 3번출구  2021   1   9  1.166667  0.933333  0.996344  14.277823   
184  송파구  올림픽공원역 3번출구  2021   1  16  6.066667  4.700000  0.997705  14.710935   
191  송파구  올림픽공원역 3번출구  2021   1  23  0.266667  0.533333  0.499999   0.999964   
193  송파구  올림픽공원역 3번출구  2021   2   1  0.100000  0.033333  1.000000  17.500000   
194  송파구  올림픽공원역 3번출구  2021   2   2  0.100000  0.000000  0.000000   0.000000   
219  송파구  올림픽공원역 3번출구  2021   3   3  0.166667  0.000000  0.000000   0.000000   
220  송파구  올림픽공원역 3번출구  2021   3   4  0.133333  0.033333  1.000000  17.666667   
221  송파구  올림픽공원역 3번출구  2021   3   5  0.933333  0.233333  1.000000  17.666667   
222  송파구  올림픽공원역 3번출구  2021   3   6  1.933333  1.133333  0.999972  16.584078   
240  송파구  올림픽공원역 3번출구  2021   4   0  1.466667  0.966667  0.999812  16.073567   
241  송파구  올림픽공원역 3번출구  2021   4   1  0.133333  0.433333  0.307692   0.444444   
242  송파구  올림픽공원역 3번출구  2021   4   2  0.066667  0.133333  0.499999   0.999964   
243  송파구  올림픽공원역 3번출구  2021   4   3  0.233333  0.066667  1.000000  17.600000   
244  송파구  올림픽공원역 3번출구  2021   4   4  0.266667  0.033333  1.000000  17.857143   
245  송파구  올림픽공원역 3번출구  2021   4   5  1.433333  0.433333  1.000000  17.566667   

            W_s       재고량  3사분위수  2사분위수  1사분위수  균형 내 균형  
30    14.531250  2.900000    6.2   4.05    2.6  균형  불균형  
53   267.000000  5.033333    6.2   4.05    2.6  균형  불균형  
54    16.219770  5.900000    6.2   4.05    2.6  균형  불균형  
61     6.247055  3.900000    6.2   4.05    2.6  균형  불균형  
64     3.179553  4.233333    6.2   4.05    2.6  균형  불균형  
78    10.528610  4.566667    6.2   4.05    2.6  균형  불균형  
96    20.021875  2.600000    6.2   4.05    2.6  균형  불균형  
97    20.513675  2.966667    6.2   4.05    2.6  균형  불균형  
98     3.749991  3.033333    6.2   4.05    2.6  균형  불균형  
100  255.001030  3.166667    6.2   4.05    2.6  균형  불균형  
101   

In [33]:
len(b)

41

In [34]:
len(b[(b['lambda'] > b['mu']) & (b['L_s'] > (20 * 0.7))])

29

- 기존방법이 균형으로 판단한 데이터 중 본 연구에서 불균형으로 판단한 데이터는 41개이지만
- 이 중 반납이 대여보다 많고 대여소 내 평균 자전거수(L_s)가 서울시 거치대율의 목표인 거치대수의 60~70%를 초과하는 경우가 29개로 70%이상의 경우에 불필요하게 자전거의 재고가 많다고 판단된다.
- 따라서 이러한 경우에는 이용자들의 자전거 이용에는 문제가 없지만 주변 대여소의 재고가 부족하다면 자전거 재배치의 측면에서는 재배치를 할 필요가 있다고 판단하여 불균형으로 판단할 수 있다.

In [35]:
df_new

자치구         대여소명     년   월  시간     lambda         mu      p_1+  \
0    송파구  올림픽공원역 3번출구  2020   6   0   2.500000   2.000000  0.996344   
1    송파구  올림픽공원역 3번출구  2020   6   1   1.033333   1.166667  0.873062   
2    송파구  올림픽공원역 3번출구  2020   6   2   0.500000   0.533333  0.911549   
3    송파구  올림픽공원역 3번출구  2020   6   3   0.433333   0.333333  0.997934   
4    송파구  올림픽공원역 3번출구  2020   6   4   0.133333   0.100000  0.998585   
5    송파구  올림픽공원역 3번출구  2020   6   5   0.733333   0.333333  1.000000   
6    송파구  올림픽공원역 3번출구  2020   6   6   3.866667   1.900000  0.999999   
7    송파구  올림픽공원역 3번출구  2020   6   7   4.366667   3.733333  0.990896   
8    송파구  올림픽공원역 3번출구  2020   6   8   7.000000   6.900000  0.953905   
9    송파구  올림픽공원역 3번출구  2020   6   9   6.066667   5.866667  0.961724   
10   송파구  올림픽공원역 3번출구  2020   6  10   5.033333   5.900000  0.845559   
11   송파구  올림픽공원역 3번출구  2020   6  11   5.933333   6.266667  0.917663   
12   송파구  올림픽공원역 3번출구  2020   6  12   7.400000   6.466667  0.987928   
13   송파구  올림픽공원역 3번출구  2020   6  13   6.766667   8.033333  0.836031   
14   송파구  올림픽공원역 3번출구  2020   6  14   6.366667   6.600000  0.928628   
15   송파구  올림픽공원역 3번출구  2020   6  15   7.633333   8.733333  0.863468   
16   송파구  올림픽공원역 3번출구  2020   6  16  11.266667  11.500000  0.937102   
17   송파구  올림픽공원역 3번출구  2020   6  17  14.600000  16.200000  0.885337   
18   송파구  올림픽공원역 3번출구  2020   6  18  16.366667  17.366667  0.914812   
19   송파구  올림픽공원역 3번출구  2020   6  19  17.600000  18.533333  0.919470   
20   송파구  올림픽공원역 3번출구  2020   6  20  21.900000  21.333333  0.958856   
21   송파구  올림픽공원역 3번출구  2020   6  21  17.233333  18.333333  0.913217   
22   송파구  올림픽공원역 3번출구  2020   6  22  12.533333  12.966667  0.929757   
23   송파구  올림픽공원역 3번출구  2020   6  23   4.633333   5.000000  0.904107   
24   송파구  올림픽공원역 3번출구  2020   7   0   1.400000   1.300000  0.975090   
25   송파구  올림픽공원역 3번출구  2020   7   1   0.700000   0.700000  0.947368   
26   송파구  올림픽공원역 3번출구  2020   7   2   0.433333   0.333333  0.997934   
27   송파구  올림픽공원역 3번출구  2020   7   3   0.266667   0.466667  0.571418   
28   송파구  올림픽공원역 3번출구  2020   7   4   0.166667   0.166667  0.947368   
29   송파구  올림픽공원역 3번출구  2020   7   5   0.466667   0.266667  0.999982   
30   송파구  올림픽공원역 3번출구  2020   7   6   3.333333   1.200000  1.000000   
31   송파구  올림픽공원역 3번출구  2020   7   7   4.166667   3.533333  0.991828   
32   송파구  올림픽공원역 3번출구  2020   7   8   4.933333   4.933333  0.947368   
33   송파구  올림픽공원역 3번출구  2020   7   9   4.600000   5.433333  0.839855   
34   송파구  올림픽공원역 3번출구  2020   7  10   4.100000   4.300000  0.921886   
35   송파구  올림픽공원역 3번출구  2020   7  11   4.400000   4.500000  0.936056   
36   송파구  올림픽공원역 3번출구  2020   7  12   5.966667   6.033333  0.941941   
37   송파구  올림픽공원역 3번출구  2020   7  13   6.666667   6.666667  0.947368   
38   송파구  올림픽공원역 3번출구  2020   7  14   7.300000   8.100000  0.885337   
39   송파구  올림픽공원역 3번출구  2020   7  15   8.300000   9.300000  0.878479   
40   송파구  올림픽공원역 3번출구  2020   7  16   9.566667   8.666667  0.981239   
41   송파구  올림픽공원역 3번출구  2020   7  17  10.233333  11.766667  0.859811   
42   송파구  올림픽공원역 3번출구  2020   7  18  12.766667  13.100000  0.934284   
43   송파구  올림픽공원역 3번출구  2020   7  19  13.266667  13.833333  0.925287   
44   송파구  올림픽공원역 3번출구  2020   7  20  15.933333  15.700000  0.954063   
45   송파구  올림픽공원역 3번출구  2020   7  21  14.200000  14.800000  0.925543   
46   송파구  올림픽공원역 3번출구  2020   7  22   8.733333   9.800000  0.877432   
47   송파구  올림픽공원역 3번출구  2020   7  23   5.133333   5.100000  0.950397   
48   송파구  올림픽공원역 3번출구  2020   8   0   0.966667   0.900000  0.974345   
49   송파구  올림픽공원역 3번출구  2020   8   1   0.833333   0.700000  0.992801   
50   송파구  올림픽공원역 3번출구  2020   8   2   0.200000   0.200000  0.947368   
51   송파구  올림픽공원역 3번출구  2020   8   3   0.166667   0.266667  0.624950   
52   송파구  올림픽공원역 3번출구  2020   8   4   0.033333   0.033333  0.947368   
53   송파구  올림픽공원역 3번출구  2020   8   5   0.400000   0.066667  1.000000   
54   송파구  올림픽공원역 3번출구  2020   8   6   1.866667   1.033333  0.999989   
55   송파구  올림픽공원역 3번출구  2020  

In [36]:
df_new.to_csv('./올림픽공원역 3번출구.csv', index = False, encoding = 'cp949')

# confusion matrix

In [23]:
len(df_new)

288

In [24]:
# 재고량이 ls보다 많거나 같을 경우 불균형 아님 = 균형임

z = df_new[df_new['L_s'] <= df_new['재고량']]
z

자치구               대여소명     년   월  시간     lambda         mu      p_1+  \
1    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   1   1.033333   1.300000  0.794608   
2    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   2   0.500000   0.666667  0.749940   
11   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  11  10.333333   8.333333  0.999530   
12   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  12  12.266667   8.766667  0.999977   
13   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  13  11.433333  12.033333  0.935505   
14   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  14  13.566667  15.266667  0.884895   
15   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  15  12.866667  17.133333  0.750911   
16   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  16  16.500000  20.166667  0.817640   
17   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  17  19.266667  28.900000  0.666664   
35   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  11   7.533333   7.000000  0.989718   
37   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  13  11.233333  10.133333  0.994244   
38   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  14  11.000000  15.633333  0.703614   
39   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  15  12.700000  15.433333  0.822271   
40   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  16  13.166667  19.166667  0.686951   
41   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  17  17.966667  23.166667  0.775398   
50   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   2   0.233333   0.433333  0.538462   
51   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   3   0.200000   0.233333  0.855489   
56   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   8   6.666667   8.733333  0.763265   
62   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  14   5.566667   7.033333  0.791232   
63   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  15   6.466667   7.500000  0.860325   
64   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  16   6.433333   8.366667  0.768811   
65   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  17   9.566667  11.500000  0.831072   
74   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9   2   0.233333   0.433333  0.538462   
85   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9  13  14.566667  14.300000  0.973692   
86   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9  14  14.100000  18.833333  0.748616   
87   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9  15  16.600000  18.833333  0.878286   
88   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9  16  18.200000  21.666667  0.838974   
89   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9  17  19.700000  25.366667  0.776463   
108  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  10  12  14.633333  13.066667  0.995331   
109  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  10  13  15.866667  16.700000  0.935477   
110  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  10  14  16.866667  21.466667  0.785517   
111  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  10  15  18.700000  19.600000  0.938294   
112  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  10  16  22.033333  26.733333  0.823542   
120  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   0   0.700000   0.500000  0.999977   
121  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   1   1.100000   0.833333  0.999898   
122  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   2   0.466667   0.466667  0.965517   
123  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   3   0.033333   0.233333  0.142857   
124  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   4   0.266667   0.300000  0.885115   
126  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   6   1.966667   0.333333  1.000000   
127  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   7   8.833333   3.500000  1.000000   
128  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   8  16.366667   9.500000  1.000000   
129  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   9  10.900000   6.900000  0.999999   
130  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  10   8.433333   4.100000  1.000000   
131  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  11   6.733333   4.966667  0.999948   
132  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  12   9.200000   8.600000  0.988505   
133  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  13   9.800000   9.633333  0.973156   
134  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  14  11.066667  11.700000  0.932411   
135  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  15  13.500000  12.000000  0.995754   
136  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  16  13.666667  17.500000  0.780784   
137  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  17  14.700000  21.833333  0.673279   
138  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  18  16.400000  25.033333  0.655125   
1

In [25]:
len(z)

89

In [26]:
# TP

len(z[z['균형'] == '균형'])

34

In [27]:
# FN

len(z[z['균형'] == '불균형'])

55

- 재고량이 ls보다 많거나 같을 경우 균형인데 불균형이라고 판단한 수가 55개

- 균형인데 균형이라고 판단한 수가 34개

In [28]:
# ls보다 재고량이 작으므로 불균형

w = df_new[df_new['L_s'] > df_new['재고량']]
w

자치구               대여소명     년   월  시간     lambda         mu      p_1+  \
0    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   0   1.666667   1.533333  0.991495   
3    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   3   0.366667   0.333333  0.993272   
4    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   4   0.833333   0.466667  1.000000   
5    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   5   1.200000   0.300000  1.000000   
6    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   6   2.933333   1.266667  1.000000   
7    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   7  10.200000   5.666667  1.000000   
8    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   8  19.033333  17.500000  0.991595   
9    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   9  17.633333   9.533333  1.000000   
10   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  10   9.733333   5.800000  1.000000   
18   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  18  27.400000  30.466667  0.894477   
19   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  19  17.966667  18.433333  0.951743   
20   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  20  14.533333  14.766667  0.957284   
21   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  21  11.633333  11.666667  0.964118   
22   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  22   7.933333   7.800000  0.973073   
23   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  23   2.966667   3.266667  0.902176   
24   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   0   1.733333   1.900000  0.905701   
25   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   1   0.900000   0.766667  0.998321   
26   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   2   0.433333   0.500000  0.864531   
27   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   3   0.633333   0.400000  0.999999   
28   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   4   0.766667   0.200000  1.000000   
29   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   5   1.266667   0.800000  0.999999   
30   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   6   2.533333   1.566667  0.999999   
31   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   7  10.033333   4.233333  1.000000   
32   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   8  15.533333  13.733333  0.996209   
33   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   9  13.333333   6.400000  1.000000   
34   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  10   7.800000   5.266667  0.999995   
36   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  12   9.700000   7.600000  0.999766   
42   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  18  21.366667  22.800000  0.925854   
43   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  19  15.400000  15.066667  0.975037   
44   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  20  11.600000  11.700000  0.961214   
45   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  21   8.366667   8.666667  0.945913   
46   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  22   5.700000   6.033333  0.931589   
47   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  23   2.800000   2.900000  0.945999   
48   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   0   1.333333   1.233333  0.990563   
49   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   1   0.900000   0.866667  0.980649   
52   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   4   0.166667   0.066667  1.000000   
53   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   5   0.733333   0.300000  1.000000   
54   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   6   1.400000   0.566667  1.000000   
55   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   7   5.866667   3.433333  1.000000   
57   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   9   7.766667   3.900000  1.000000   
58   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  10   5.433333   4.166667  0.999862   
59   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  11   5.566667   5.100000  0.992158   
60   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  12   5.266667   4.366667  0.999097   
61   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  13   5.933333   5.133333  0.997628   
66   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  18  10.333333  10.700000  0.946138   
67   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  19   8.800000   8.466667  0.980928   
68   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  20   6.733333   7.233333  0.920976   
69   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  21   5.100000   5.633333  0.899721   
70   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  22   3.733333   3.700000  0.969672   
71   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  23   1.833333   2.066667  0.883487   
72   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9   0   2.600000   2.300000  0.996164   
7

In [29]:
len(w)

199

In [30]:
# FP

len(w[w['균형'] == '균형'])

111

In [31]:
# TN

len(w[w['균형'] == '불균형'])

88

- 불균형인데 균형이라고 판단한 수는 111
- 불균형인데 불균형이라고 판단한 수는 88

# confusion matrix

0 |균형|불균형
-|-|-
균형|TP|FP
불균형|FN|TN


1. 정확도 accuracy = 실제(내) 데이터와 예측(기존방법)데이터가 얼마나 같은지를 평가하는 지표 = TP + TN / TP+TN+FP+FN
2. 정밀도 precision = 균형으로 예측한 데이터 중 실제로 균형인 데이터 비율 = TP / TP+FP
3. 재현율 recall = 실제 균형 중 균형으로 예측한 비율 = TP / TP+FN

=> 정밀도에 초점. (불균형을 균형으로 예측하여 재배치하지 않으면 서비스에 차질)


In [32]:
# 정확도

(len(z[z['균형'] == '균형']) + len(w[w['균형'] == '불균형'])) / len(df_new)

0.4236111111111111

In [33]:
# 정밀도

len(z[z['균형'] == '균형']) / (len(z[z['균형'] == '균형']) + len(w[w['균형'] == '균형']))

0.23448275862068965

In [34]:
# 재현율

len(z[z['균형'] == '균형']) / (len(z[z['균형'] == '균형']) + len(z[z['균형'] == '불균형']))

0.38202247191011235